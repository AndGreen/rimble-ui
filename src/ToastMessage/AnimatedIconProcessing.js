import React from 'react';
import styled from '@emotion/styled';
import { keyframes } from '@emotion/core';
import { ReactComponent as IconProcessing } from './icon-processing.svg';

const flicker = keyframes`
  0% {
    opacity: 0.1;
    transform: translateY(-1px);
  }
  20% {
    opacity: 0.6;
  }
  40% {
    opacity: 0.9;
    transform: translateY(1px);
  }
  60% {
    opacity: 0.5;
    transform: translateY(2px);
  }
  100% {
    opacity: 0;
    transform: translateY(3px);
  }
`;

const StyledIcon = styled(IconProcessing)`
  & {
    display: block;
  }
  > * {
    animation-name: ${flicker};
    animation-duration: ${props => props.seeds.seed + 'ms'};
    animation-timing-function: ease;
    animation-delay: 0s;
    animation-iteration-count: infinite;
  }
  > *:nth-of-type(4n + 0) {
    animation-duration: ${props => props.seeds.seeda + 'ms'};
    animation-delay: ${props => props.seeds.seeda + 'ms'};
  }
  > *:nth-of-type(2n + 0) {
    animation-duration: ${props => props.seeds.seedb + 'ms'};
  }
  > *:nth-of-type(3n + 0) {
    animation-duration: ${props => props.seeds.seeda + 'ms'};
    animation-delay: ${props => props.seeds.seedb + 'ms'};
  }
`;

const AnimatedIconProcessing = props => {
  const seeds = {
    seed: Math.floor(1000 + Math.random() * 7000),
    seeda: Math.floor(
      Date.now()
        .toString()
        .substring(2, 6) +
        Math.random() * 9000
    )
      .toString()
      .substring(0, 4),
    seedb: Math.floor(
      Date.now()
        .toString()
        .substring(0, 4) +
        Math.random() * 4500
    )
      .toString()
      .substring(0, 4),
  };

  return <StyledIcon seeds={seeds} {...props} />;
};

export default AnimatedIconProcessing;
