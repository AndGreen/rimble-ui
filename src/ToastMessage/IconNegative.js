import React, { Component } from 'react';
import styled from '@emotion/styled';
import { ReactComponent as NegativeIcon } from './icon-negative.svg';

const WrappedIconNegative = props => <NegativeIcon {...props} />;

const IconNegative = styled(WrappedIconNegative)``;

export default IconNegative;
