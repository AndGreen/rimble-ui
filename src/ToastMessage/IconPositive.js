import React, { Component } from 'react';
import styled from '@emotion/styled';
import { ReactComponent as PositiveIcon } from './icon-positive.svg';

const WrappedIconPositive = props => <PositiveIcon {...props} />;

const IconPositive = styled(WrappedIconPositive)``;

export default IconPositive;
