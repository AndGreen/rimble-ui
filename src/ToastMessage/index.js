import React, { Component } from 'react';
import styled from 'styled-components'

import Box from '../Box'
import Icon from '../Icon'
import Text from '../Text'
import TextButton from '../TextButton'

import { ReactComponent as IconPositive } from './icon-positive.svg'
import { ReactComponent as IconNegative } from './icon-negative.svg'
import { ReactComponent as IconProcessing } from './icon-processing.svg'


const getColors = (props) => {
  switch (props.variant) {
    case 'dark':
      return `
        color: #FFF;
        background-color: #000;
        button {color: #9387FF}
        svg:first-child {fill: #FFF}
      `;
      break;
    default:
      return`
        color: #000;
        background-color: #FFF;
        // svg {fill: #999999}
      `;
  }
}

const StyledToastContainer = styled('div')`
  & {
    display: block;
    position: fixed;
    top: auto;
    bottom: 0;
    left: auto;
    right: 0;
    width: 480px;
    pointer-events: none;
    padding: 1rem;
  }
`

const StyledToastMessage = styled(Box)`
  & {
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    user-select: none;
    height: 80px;

    ${'' /* background: #FFFFFF; */}
    border: 1px solid #D6D6D6;
    box-shadow: 0px 8px 16px rgba(0, 0, 0, 0.1);
    border-radius: 4px;

    pointer-events: all;
  }
  &:hover {

  }

  ${getColors}
`



const ToastMessage = ({className, ...props}) => (
  <StyledToastMessage className={className} {...props}>
    { props.icon && <Icon name={props.icon} color={'primary'} size={'32px'} mr={2} flex={'0 0'} /> }
    <Box flex={'1 0'} mx={2}>
      { props.message && <Text fontSize={1} fontWeight={3} color={'inherit'}>{props.message}</Text> }
      { props.secondaryMessage && <Text fontSize={1} color={'#666'}>{props.secondaryMessage}</Text> }
    </Box>
    <TextButton flex={'0 0 auto'} p={'0'} color={'primary'}>
      { props.actionText }
    </TextButton>
    <TextButton icononly size={'small'} alignSelf={'flex-start'} >
      <Icon name={'Close'} size={'16px'} color={'grey'} flex={'0 0'} />
    </TextButton>
  </StyledToastMessage>
);

const SuccessToastMessage = ({className, ...props}) => (
  <StyledToastMessage className={className} {...props}>
    <Box size={'32px'} mr={2} flex={'0 0'} >
      <IconPositive width={'32px'} height={'32px'} />
    </Box>
    <Box flex={'1 0'} mx={2}>
      { props.message && <Text fontSize={1} fontWeight={3} color={'inherit'}>{props.message}</Text> }
      { props.secondaryMessage && <Text fontSize={1} color={'#666'}>{props.secondaryMessage}</Text> }
    </Box>
    <TextButton flex={'0 0 auto'} p={'0'} color={'primary'}>
      { props.actionText }
    </TextButton>
    <TextButton icononly size={'small'} alignSelf={'flex-start'} >
      <Icon name={'Close'} size={'16px'} color={'grey'} flex={'0 0'} />
    </TextButton>
  </StyledToastMessage>
)

const FailureToastMessage = ({className, ...props}) => (
  <StyledToastMessage className={className} {...props}>
    <Box size={'32px'} mr={2} flex={'0 0'} >
      <IconNegative width={'32px'} height={'32px'} />
    </Box>
    <Box flex={'1 0'} mx={2}>
      { props.message && <Text fontSize={1} fontWeight={3} color={'inherit'}>{props.message}</Text> }
      { props.secondaryMessage && <Text fontSize={1} color={'#666'}>{props.secondaryMessage}</Text> }
    </Box>
    <TextButton flex={'0 0 auto'} p={'0'} color={'primary'}>
      { props.actionText }
    </TextButton>
    <TextButton icononly size={'small'} alignSelf={'flex-start'} >
      <Icon name={'Close'} size={'16px'} color={'grey'} flex={'0 0'} />
    </TextButton>
  </StyledToastMessage>
)

const ProcessingToastMessage = React.forwardRef((props, ref) => (
  <StyledToastMessage ref={ref} {...props}>
    <Box size={'32px'} mr={2} flex={'0 0'} >
      <IconProcessing width={'32px'} height={'32px'} />
    </Box>
    <Box flex={'1 0'} mx={2}>
      { props.message && <Text fontSize={1} fontWeight={3} color={'inherit'}>{props.message}</Text> }
      { props.secondaryMessage && <Text fontSize={1} color={'#666'}>{props.secondaryMessage}</Text> }
    </Box>
    <TextButton onClick={() => (console.log('Action link!'))} color={'primary'} p={'0'} flex={'0 0 auto'}>
      { props.actionText }
    </TextButton>
    <TextButton onClick={() => (console.log('Close me!'))} icononly size={'small'} alignSelf={'flex-start'} >
      <Icon name={'Close'} size={'16px'} color={'grey'} flex={'0 0'} />
    </TextButton>
  </StyledToastMessage>
));

// wip
class ProtoToastMessage extends Component {
  constructor(props) {
    super(props);
    this.toastNode = React.createRef();
  }

  handleClose = (e) => {
    e.preventDefault();
    console.log('close!');
    console.log(this.toastNode.current);
  }

  handleAction = (e) => {
    e.preventDefault();
    console.log('action!');
  }

  render() {
    const props = this.props;
    return (
      <StyledToastMessage ref={this.toastNode}>
        <Box size={'32px'} mr={2} flex={'0 0'} >
          <IconProcessing width={'32px'} height={'32px'} />
        </Box>
        <Box flex={'1 0'} mx={2}>
          { props.message && <Text fontSize={1} fontWeight={3} color={'inherit'}>{props.message}</Text> }
          { props.secondaryMessage && <Text fontSize={1} color={'#666'}>{props.secondaryMessage}</Text> }
        </Box>
        <TextButton onClick={this.handleAction} color={'primary'} p={'0'} flex={'0 0 auto'}>
          { props.actionText }
        </TextButton>
        <TextButton onClick={this.handleClose} icononly size={'small'} alignSelf={'flex-start'} >
          <Icon name={'Close'} size={'16px'} color={'grey'} flex={'0 0'} />
        </TextButton>
      </StyledToastMessage>
    );
  }

}

class ToastContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false
    }
    this.closeBttn = React.createRef();
  }

  handleClose = (e) => {
    e.preventDefault();
    // this.closeBttn.current.remove()
    console.log('close!', this.closeBttn);
  }

  handleAction = (e) => {
    e.preventDefault();
    console.log('action!');
  }

  render() {
    return (
      <StyledToastContainer p={3}>

        <ProtoToastMessage
          message={'Processing Paymentâ€¦'}
          secondaryMessage={''}
          actionText={'View Details'}

          mt={2}
         />

      </StyledToastContainer>
    )
  }
};

ToastMessage.Container = ToastContainer;

ToastMessage.Success = SuccessToastMessage;
ToastMessage.Failure = FailureToastMessage;
ToastMessage.Processing = ProcessingToastMessage;

StyledToastMessage.defaultProps = {
  display: 'flex',
  flexDirection: 'row nowrap',
  alignItems: 'center',
  borderRadius: 1,
  py: 0,
  pl: 3,
  maxWidth: 6,
}

ToastMessage.defaultProps = {
  message:'Generic Message',
  secondaryMessage: '',
  actionText: 'Action',
  icon: false
}

ToastMessage.displayName = 'ToastMessage'

export default ToastMessage;
