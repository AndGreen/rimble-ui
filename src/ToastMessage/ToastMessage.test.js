import ToastMessage from './'

describe('ToastMessage component sanity', () => {
  it('has name', () => {
    expect(ToastMessage.displayName).toBe('ToastMessage')
  })

  it('matches generic snapshot', () => {
    const toastMessage = render(<ToastMessage />)
    expect(toastMessage).toMatchSnapshot()
  })

  it('matches processing snapshot', () => {
    const toastMessage = render(<ToastMessage variant={'processing'} />)
    expect(toastMessage).toMatchSnapshot()
  })

  it('matches success snapshot', () => {
    const toastMessage = render(<ToastMessage variant={'success'} />)
    expect(toastMessage).toMatchSnapshot()
  })

  it('matches failure snapshot', () => {
    const toastMessage = render(<ToastMessage variant={'failure'} />)
    expect(toastMessage).toMatchSnapshot()
  })

  it('matches full component snapshot', () => {
    const toastMessage = render(<ToastMessage variant={'default'} message={'message'} secondaryMessage={'secondary message'} actionHref={'http://test.com'} actionText={'Link'} icon={'InfoOutline'} />)
    expect(toastMessage).toMatchSnapshot()
  })
})

describe('ToastMessage component unit tests', () => {
  it('displays secondary message', () => {
    const toastMessage = mount(
      <ToastMessage message={''} secondaryMessage={'Lorem ipsum dolor sit.'} />,
    )
    expect(toastMessage.children().text()).toBe('Lorem ipsum dolor sit.')
    toastMessage.unmount()
  })

  it('displays icon', () => {

  })

  it('accepts action text', () => {

  })

  it('accepts action href', () => {

  })

  it('displays as processing', () => {

  })

  it('displays as success', () => {

  })

  it('displays as failure', () => {

  })

  it('displays as dark variant of generic', () => {

  })

  it('displays as dark variant of default', () => {

  })

  it('accepts proxy props', () => {
    const onClick = jest.fn()
    const toastMessage = mount(
      <ToastMessage onClick={onClick} />,
    )

    expect(toastMessage.at(0).props().onClick).toBe(onClick)

    toastMessage.unmount()
  })

})