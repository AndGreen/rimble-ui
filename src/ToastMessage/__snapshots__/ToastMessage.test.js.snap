// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastMessage component sanity matches failure snapshot 1`] = `
.c1 {
  margin-left: 8px;
  margin-right: 8px;
  -webkit-flex: 1 0;
  -ms-flex: 1 0;
  flex: 1 0;
}

.c1 {
  box-sizing: border-box;
}

.c3 {
  margin-right: 8px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c3 {
  box-sizing: border-box;
}

.c2 {
  color: inherit;
  font-size: 14px;
  font-size: 14px;
  font-family: sansSerif;
  font-weight: 3;
  text-align: left;
  line-height: copy;
}

.c2 {
  box-sizing: border-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row nowrap;
  -ms-flex-direction: row nowrap;
  flex-direction: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  background-color: #FFF;
}

.c0 {
  box-sizing: border-box;
}

.c0 {
  pointer-events: all;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 80px;
  padding: 0 1rem;
  box-shadow: 0px 8px 16px rgba(0,0,0,0.1);
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c0:hover {
  box-shadow: 0px 8px 16px rgba(0,0,0,0.15);
}

.c0 > .iconBox {
  display: none;
}

.c0 > .closeBttn {
  display: none;
}

@media screen and (min-width:420px) {
  .c0 {
    border: 1px solid #D6D6D6;
    border-radius: 4px;
  }

  .c0 > .iconBox {
    display: block;
  }

  .c0 > .closeBttn {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      color="inherit"
      font-family="sansSerif"
      font-size="1"
      font-weight="3"
    >
      Generic message
    </div>
  </div>
  <div
    class="c3"
  >
    
  </div>
</div>
`;

exports[`ToastMessage component sanity matches full component snapshot 1`] = `
.c3 {
  margin-left: 8px;
  margin-right: 8px;
  -webkit-flex: 1 0;
  -ms-flex: 1 0;
  flex: 1 0;
}

.c3 {
  box-sizing: border-box;
}

.c6 {
  margin-right: 8px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c6 {
  box-sizing: border-box;
}

.c1 {
  margin-right: 8px;
  -webkit-flex: 0 0;
  -ms-flex: 0 0;
  flex: 0 0;
}

.c1 {
  box-sizing: border-box;
}

.c2 {
  color: primary;
}

.c4 {
  color: inherit;
  font-size: 14px;
  font-size: 14px;
  font-family: sansSerif;
  font-weight: 3;
  text-align: left;
  line-height: copy;
}

.c4 {
  box-sizing: border-box;
}

.c5 {
  color: #666;
  font-size: 14px;
  font-size: 14px;
  font-family: sansSerif;
  font-weight: 2;
  text-align: left;
  line-height: copy;
}

.c5 {
  box-sizing: border-box;
}

.c7 {
  color: primary;
  font-size: 14px;
  font-size: 14px;
  font-weight: 3;
}

.c7 {
  box-sizing: border-box;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7:hover {
  color: #6755F3;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7:active {
  color: #3223A9;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row nowrap;
  -ms-flex-direction: row nowrap;
  flex-direction: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  background-color: #FFF;
}

.c0 {
  box-sizing: border-box;
}

.c0 {
  pointer-events: all;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 80px;
  padding: 0 1rem;
  box-shadow: 0px 8px 16px rgba(0,0,0,0.1);
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c0:hover {
  box-shadow: 0px 8px 16px rgba(0,0,0,0.15);
}

.c0 > .iconBox {
  display: none;
}

.c0 > .closeBttn {
  display: none;
}

@media screen and (min-width:420px) {
  .c0 {
    border: 1px solid #D6D6D6;
    border-radius: 4px;
  }

  .c0 > .iconBox {
    display: block;
  }

  .c0 > .closeBttn {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  class="c0"
  display="flex"
>
  <div
    class="iconBox c1"
  >
    <svg
      class="c2"
      color="primary"
      display="block"
      fill="currentcolor"
      height="32px"
      viewBox="0 0 24 24"
      width="32px"
    >
      <path
        d="M11 17h2v-6h-2v6zm1-15C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zM11 9h2V7h-2v2z"
      >
        
      </path>
    </svg>
  </div>
  <div
    class="c3"
  >
    <div
      class="c4"
      color="inherit"
      font-family="sansSerif"
      font-size="1"
      font-weight="3"
    >
      message
    </div>
    <div
      class="c5"
      color="#666"
      font-family="sansSerif"
      font-size="1"
      font-weight="2"
    >
      secondary message
    </div>
  </div>
  <div
    class="c6"
  >
    <a
      class="c7"
      color="primary"
      font-size="1"
      font-weight="3"
      href="http://test.com"
      target="_blank"
    >
      Link
    </a>
  </div>
</div>
`;

exports[`ToastMessage component sanity matches generic snapshot 1`] = `
.c1 {
  margin-left: 8px;
  margin-right: 8px;
  -webkit-flex: 1 0;
  -ms-flex: 1 0;
  flex: 1 0;
}

.c1 {
  box-sizing: border-box;
}

.c3 {
  margin-right: 8px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c3 {
  box-sizing: border-box;
}

.c2 {
  color: inherit;
  font-size: 14px;
  font-size: 14px;
  font-family: sansSerif;
  font-weight: 3;
  text-align: left;
  line-height: copy;
}

.c2 {
  box-sizing: border-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row nowrap;
  -ms-flex-direction: row nowrap;
  flex-direction: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  background-color: #FFF;
}

.c0 {
  box-sizing: border-box;
}

.c0 {
  pointer-events: all;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 80px;
  padding: 0 1rem;
  box-shadow: 0px 8px 16px rgba(0,0,0,0.1);
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c0:hover {
  box-shadow: 0px 8px 16px rgba(0,0,0,0.15);
}

.c0 > .iconBox {
  display: none;
}

.c0 > .closeBttn {
  display: none;
}

@media screen and (min-width:420px) {
  .c0 {
    border: 1px solid #D6D6D6;
    border-radius: 4px;
  }

  .c0 > .iconBox {
    display: block;
  }

  .c0 > .closeBttn {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      color="inherit"
      font-family="sansSerif"
      font-size="1"
      font-weight="3"
    >
      Generic message
    </div>
  </div>
  <div
    class="c3"
  >
    
  </div>
</div>
`;

exports[`ToastMessage component sanity matches processing snapshot 1`] = `
.c1 {
  margin-left: 8px;
  margin-right: 8px;
  -webkit-flex: 1 0;
  -ms-flex: 1 0;
  flex: 1 0;
}

.c1 {
  box-sizing: border-box;
}

.c3 {
  margin-right: 8px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c3 {
  box-sizing: border-box;
}

.c2 {
  color: inherit;
  font-size: 14px;
  font-size: 14px;
  font-family: sansSerif;
  font-weight: 3;
  text-align: left;
  line-height: copy;
}

.c2 {
  box-sizing: border-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row nowrap;
  -ms-flex-direction: row nowrap;
  flex-direction: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  background-color: #FFF;
}

.c0 {
  box-sizing: border-box;
}

.c0 {
  pointer-events: all;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 80px;
  padding: 0 1rem;
  box-shadow: 0px 8px 16px rgba(0,0,0,0.1);
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c0:hover {
  box-shadow: 0px 8px 16px rgba(0,0,0,0.15);
}

.c0 > .iconBox {
  display: none;
}

.c0 > .closeBttn {
  display: none;
}

@media screen and (min-width:420px) {
  .c0 {
    border: 1px solid #D6D6D6;
    border-radius: 4px;
  }

  .c0 > .iconBox {
    display: block;
  }

  .c0 > .closeBttn {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      color="inherit"
      font-family="sansSerif"
      font-size="1"
      font-weight="3"
    >
      Generic message
    </div>
  </div>
  <div
    class="c3"
  >
    
  </div>
</div>
`;

exports[`ToastMessage component sanity matches success snapshot 1`] = `
.c1 {
  margin-left: 8px;
  margin-right: 8px;
  -webkit-flex: 1 0;
  -ms-flex: 1 0;
  flex: 1 0;
}

.c1 {
  box-sizing: border-box;
}

.c3 {
  margin-right: 8px;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
}

.c3 {
  box-sizing: border-box;
}

.c2 {
  color: inherit;
  font-size: 14px;
  font-size: 14px;
  font-family: sansSerif;
  font-weight: 3;
  text-align: left;
  line-height: copy;
}

.c2 {
  box-sizing: border-box;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row nowrap;
  -ms-flex-direction: row nowrap;
  flex-direction: row nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #000;
  background-color: #FFF;
}

.c0 {
  box-sizing: border-box;
}

.c0 {
  pointer-events: all;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 80px;
  padding: 0 1rem;
  box-shadow: 0px 8px 16px rgba(0,0,0,0.1);
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c0:hover {
  box-shadow: 0px 8px 16px rgba(0,0,0,0.15);
}

.c0 > .iconBox {
  display: none;
}

.c0 > .closeBttn {
  display: none;
}

@media screen and (min-width:420px) {
  .c0 {
    border: 1px solid #D6D6D6;
    border-radius: 4px;
  }

  .c0 > .iconBox {
    display: block;
  }

  .c0 > .closeBttn {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

<div
  class="c0"
  display="flex"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      color="inherit"
      font-family="sansSerif"
      font-size="1"
      font-weight="3"
    >
      Generic message
    </div>
  </div>
  <div
    class="c3"
  >
    
  </div>
</div>
`;
